app/src/main/java/org/thoughtcrime/securesms/conversation/v2/messages/MessageUtilities.kt

```diff
@@ -0,0 +1,15 @@
+package org.thoughtcrime.securesms.conversation.v2.messages
+
+import android.widget.TextView
+import androidx.core.view.isVisible
+import org.thoughtcrime.securesms.database.model.MessageRecord
+import org.thoughtcrime.securesms.util.DateUtils
+import java.util.Locale
+
+private const val maxTimeBetweenBreaks = 5 * 60 * 1000L // 5 minutes
+
+fun TextView.showDateBreak(message: MessageRecord, previous: MessageRecord?) {
+    val showDateBreak = (previous == null || message.timestamp - previous.timestamp > maxTimeBetweenBreaks)
+    isVisible = showDateBreak
+    text = if (showDateBreak) DateUtils.getDisplayFormattedTimeSpanString(context, Locale.getDefault(), message.timestamp) else ""
```

Comment:
Should we use a `null` for the setting of text instead of empty string? I realise it might be optimised away but that could be an improvement over how it used to be done.

Reply:
`TextView.setText` actually checks for nulls and replaces with an empty string so we could just keep as is?